dynamiqs/mesolve/mesolve.py

```diff
@@ -167,6 +161,8 @@ def broadcast(x: TimeArray) -> TimeArray:
         f = _cartesian_vectorize(_mesolve, n_batch, out_axes)
     else:
         f = _flat_vectorize(_mesolve, n_batch, out_axes)
+        H = H.squeeze()
+        jump_ops = [x.squeeze() for x in jump_ops]
```

Comment:
I don't see how this could work. The output _shape_ will be correct, but the computation will be the same for H `(2, 1, n, n)` and `(1, 2, n, n)` (it will always be squeeze to `(2, n, n)`.

See previous comment
> If the passed Hamiltonian is of shape (1, n, n) or (2, 1, n, n) we don't want to squeeze the one. IMO the magic you propose should be done by _flat_vectorize, it has enough information with the shape of the objects.